Mage/src/main/java/mage/target/targetadjustment/TargetAdjuster.java

```diff
@@ -11,5 +11,8 @@
 @FunctionalInterface
 public interface TargetAdjuster extends Serializable {
 
+    // Please note that if refactoring and a copy() method is to be added, you must also
```

Comment:
Delete it and add info like “warning, do not use changeable data inside, it must be static objects like Filter”

Reply:
Just to be clear: Target _in general_ is changeable, but `EachOpponentPermanentTargetsAdjuster` takes that Target object, makes a unique copy for its own use, and then never changes it. Whenever `adjustTargets` is called it makes a copy of the blueprint, so the original blueprint remains safe/immutable.

Reply:
Yea, same logic with Filter